chapter_linear-regression/oo-design.md

```diff
@@ -281,15 +332,42 @@ class Trainer(d2l.HyperParameters):  #@save
         model.board.xlim = [0, self.max_epochs]
         self.model = model
 
-    def fit(self, model, data):
-        self.prepare_data(data)
-        self.prepare_model(model)
-        self.optim = model.configure_optimizers()
-        self.epoch = 0
-        self.train_batch_idx = 0
-        self.val_batch_idx = 0
-        for self.epoch in range(self.max_epochs):
-            self.fit_epoch()
+    if tab.selected('pytorch', 'mxnet', 'tensorflow'):
+        def fit(self, model, data):
+            self.prepare_data(data)
+            self.prepare_model(model)
+            self.optim = model.configure_optimizers()
+            self.epoch = 0
+            self.train_batch_idx = 0
+            self.val_batch_idx = 0
+            for self.epoch in range(self.max_epochs):
+                self.fit_epoch()
+
+    if tab.selected('jax'):
+        def apply_init(self, **kwargs):
+            if kwargs and 'key' in kwargs and (kwargs['key'] is not None):
+                self.key = kwargs['key']
+            else:
+                self.key = jax.random.PRNGKey(0)  # Avoid, but use as fallback
+            input_shape = next(iter(self.train_dataloader))[0].shape
+            dummy_input = jnp.zeros(input_shape)
+            params = self.model.init(self.key, dummy_input)
+            return params
+
+        def fit(self, model, data, key=None):
+            self.prepare_data(data)
+            self.prepare_model(model)
+            self.params = self.apply_init(key=key)
```

Comment:
Consider defining a d2l.SEED constant to be fed into key=jax.random.PRNGKey(d2l.SEED) here. So we can simplify usage of this method like:

```
key = jax.random.PRNGKey(24)
trainer.fit(model, data, key)
```
as
```
trainer.fit(model, data)
```

Reply:
See comment above.